packages/cloudflare/src/cli/build/utils/extract-project-env-vars.ts

```diff
@@ -0,0 +1,34 @@
+import * as fs from "node:fs";
+import * as path from "node:path";
+
+import { parse } from "@dotenvx/dotenvx";
+import type { BuildOptions } from "@opennextjs/aws/build/helper.js";
+
+function readEnvFiles(fileNames: string[], { monorepoRoot, appPath }: BuildOptions) {
+  return fileNames
+    .flatMap((fileName) => [
+      ...(monorepoRoot !== appPath ? [path.join(monorepoRoot, fileName)] : []),
+      path.join(appPath, fileName),
+    ])
+    .filter((filePath) => fs.existsSync(filePath) && fs.statSync(filePath).isFile())
+    .map((filePath) => parse(fs.readFileSync(filePath).toString()))
+    .reduce<Record<string, string>>((acc, overrides) => ({ ...acc, ...overrides }), {});
+}
+
+/**
+ * Extracts the environment variables defined in various .env files for a project.
+ *
+ * The `NEXTJS_ENV` environment variable in `.dev.vars` determines the mode.
+ *
+ * Merged variables respect the following priority order.
+ * 1. `.env.{mode}.local`
+ * 2. `.env.local`
+ * 3. `.env.{mode}`
+ * 4. `.env`
+ *
+ * In a monorepo, the env files in an app's directory will take precedence over
+ * the env files at the root of the monorepo.
+ */
+export function extractProjectEnvVars(mode: string, options: BuildOptions) {
```

Comment:
Doesn't that mean that we will always use the `.local` files while we only want to use them for `wrangler dev` but not in prod?

Reply:
Yeah. How come you wouldn't want them in production? Next.js would use those files. I suppose you're talking about if someone builds it locally and then deploys, but in that case, you could make the argument that we shouldnt include any .env files at all because technically they're all local to the machine. I'm not quite sure what else you could do here - open to suggestions though.

Reply:
@james-elicx I did a quick test and you are right, `.local` files are used if present, regardless if it is a prod or dev build. I was wrong to think that they would only be used in dev.